libsession/src/main/java/org/session/libsession/messaging/file_server/FileServerApi.kt

```diff
@@ -105,4 +110,52 @@ object FileServerApi {
         val request = Request(verb = HTTP.Verb.GET, endpoint = "file/$file")
         return send(request)
     }
+
+    /**
+     * Returns the current version of session
+     * This is effectively proxying (and caching) the response from the github release
+     * page.
+     *
+     * Note that the value is cached and can be up to 30 minutes out of date normally, and up to 24
+     * hours out of date if we cannot reach the Github API for some reason.
+     *
+     * https://github.com/oxen-io/session-file-server/blob/dev/doc/api.yaml#L119
+     */
+    suspend fun getClientVersion(): VersionData {
+        // Generate the auth signature
+        val secretKey =  MessagingModuleConfiguration.shared.getUserED25519KeyPair()?.secretKey?.asBytes
+            ?: throw (Error.NoEd25519KeyPair)
+
+        val blindedKeys = BlindKeyAPI.blindVersionKeyPair(secretKey)
+        val timestamp = TimeUnit.MILLISECONDS.toSeconds(System.currentTimeMillis()) //  The current timestamp in seconds
```

Comment:
so... you could perhaps more clearly:

```
val timestampSeconds = System.currentTimeMillis().milliseconds.inWholeSeconds
```

as it's useful to put units in the variable name if it is not contained in the type.

alternatively

```
val timestamp = System.currentTimeMillis().milliseconds
val signature = BlindKeyAPI.blindVersionSign(secretKey, timestamp.inWholeSeconds)
...
"X-FS-Timestamp" to timestamp.inWholeSeconds.toString(),
```